modules/host/negotiate.go

```diff
@@ -231,6 +238,9 @@ func (h *Host) rpcUpload(conn net.Conn) error {
 	err = h.tpool.AcceptTransactionSet(signedTxnSet)
 	if err == modules.ErrDuplicateTransactionSet {
 		// this can happen if the host is uploading to itself
+		//
+		// TODO: is it possible for renter to cause a collision, overwriting a
+		// previous file contract?
```

Comment:
doesn't seem likely? what is the data that might be colliding?

Assuming that there's a siacoin output being used somewhere, you've got a guarantee that siacoin outputs will never collide. The IDs of transactions could collide though, because it's just a hash of the contents. But presumably, if you've made a colliding transaction you've already put all of the interesting information on the blockchain.